processor/tailsamplingprocessor/processor.go

```diff
@@ -72,15 +72,16 @@ func newTracesProcessor(ctx context.Context, settings component.TelemetrySetting
 		return nil, component.ErrNilNextConsumer
 	}
 
-	numDecisionBatches := math.Max(1, cfg.DecisionWait.Seconds())
-	inBatcher, err := idbatcher.New(uint64(numDecisionBatches), cfg.ExpectedNewTracesPerSec, uint64(2*runtime.NumCPU()))
-	if err != nil {
-		return nil, err
-	}
-
+	policyNames := map[string]bool{}
 	policies := make([]*policy, len(cfg.PolicyCfgs))
 	for i := range cfg.PolicyCfgs {
 		policyCfg := &cfg.PolicyCfgs[i]
+
+		if policyNames[policyCfg.Name] {
+			return nil, fmt.Errorf("duplicate policy name %q", policyCfg.Name)
```

Comment:
```suggestion
			return nil, fmt.Errorf("Duplicate policy names are not allowed. Duplicate name: %q", policyCfg.Name)
```
Just a suggestion, feel free to ignore. (The test would need to be updated as well if this is accepted)

Reply:
Errors should not start with an uppercase: https://google.github.io/styleguide/go/decisions.html#error-strings